Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM --platform=linux/amd64 ubuntu:20.04 as builder
+
+RUN apt-get update
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y clang cmake make
+
+ADD . /repo
+WORKDIR /repo/build
+RUN cmake ../src
+RUN make -j8
+RUN ldconfig
```

Comment:
Are the `ldd` stuff necessary? How about explicitly copying the libraries you need? Something like:

```

FROM --platform=linux/amd64 ubuntu:20.04 as fuzzer

COPY --from=builder /repo/build/buzz/bzzasm /bzzasm
COPY --from=builder /repo/build/buzz/libbuzz.so /usr/lib/x86_64-linux-gnu/
COPY --from=builder /repo/build/buzz/libbuzzdbg.so /usr/lib/x86_64-linux-gnu/
```

Would probably be better. That being said, that still seems very manual and fragile. Does the project come with an automatic/maintained way to install? Sometimes there is a `make install` or similar where you can specify a prefix for files in your installation like ./configure --prefix=/install and then your fuzzer stage could become a trivial directory copy.

Reply:
I agree that the explicit approach is more fragile than using `ldd` directly. In this case, the target binary does not depend on any shared libraries other than those built in the `builder` stage and some standard libraries. However, I have found that other targets often have external dependencies (e.g. some library `install`ed by `apt`), which can only be captured by `ldd`. One could `apt install` them in the `package` stage, but that would pull down unnecessary transitive dependencies, header files, package lists, etc. which would substantially increase the size of the resulting image. In short, I think using `ldd` is the most robust, general solution, although in this case specifically there may be a cleaner alternative. Thoughts?

Reply:
> One could apt install them in the package stage, but that would pull down unnecessary transitive dependencies, header files, package lists, etc.

That is exactly true. And having those transitive dependencies ensures that the starting library will be able to function properly (including configuration files, dynamically loaded libraries and so on). Realistically, this is also the only way a production environment with that target would run. However, you are saying for this particular target there are no external dependencies so that doesn't quite apply? Or am I misunderstanding?

How about a `make install` or other similar method here?

Reply:
I spoke with Alex about this - I was focused on optimizing image size rather than mimicking a prod environment (i.e. rather than removing only build-time dependencies and the source repository from the builder stage).

Yes, for this target there are no external run-time dependencies. If there were, I would `apt install` them in the package stage, which I think is the approach you are looking for?

Reply:
Just saw your latest push. Phenomenal, that's exactly what I was hoping for!

> If there were, I would apt install them in the package stage, which I think is the approach you are looking for?
Exactly